src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -36,19 +38,29 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
     protected $dispatcher;
     protected $resolver;
     protected $requestStack;
+    private $argumentResolver;
 
     /**
      * Constructor.
      *
-     * @param EventDispatcherInterface    $dispatcher   An EventDispatcherInterface instance
-     * @param ControllerResolverInterface $resolver     A ControllerResolverInterface instance
-     * @param RequestStack                $requestStack A stack for master/sub requests
+     * @param EventDispatcherInterface    $dispatcher       An EventDispatcherInterface instance
+     * @param ControllerResolverInterface $resolver         A ControllerResolverInterface instance
+     * @param RequestStack                $requestStack     A stack for master/sub requests
+     * @param ArgumentResolverInterface   $argumentResolver An ArgumentResolverInterface instance
      */
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null)
```

Comment:
If there's no call to the parent constructor, they're breaking the design in the first place. The only way they would get it to work is via the protected variables and I consider this a case that shouldn't be supported, the constructor is there for a reason.